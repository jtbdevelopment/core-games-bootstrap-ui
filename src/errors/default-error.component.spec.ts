import {TestBed} from '@angular/core/testing';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {DefaultErrorComponent} from './default-error.component';
import {PlayerService} from 'jtb-core-games-ui';

class MockModal {
    dismiss = jasmine.createSpy('dismiss');
}

class MockPlayerService {
    forceLogout = jasmine.createSpy('forceLogout');
}

describe('Component:  default error component', () => {
    let fixture;
    let modal;
    let playerService;

    beforeEach(() => {
        TestBed.configureTestingModule({
            declarations: [
                DefaultErrorComponent,
            ],
            providers: [
                {provide: NgbActiveModal, useClass: MockModal},
                {provide: PlayerService, useClass: MockPlayerService}
            ],
        });
        TestBed.compileComponents();
        fixture = TestBed.createComponent(DefaultErrorComponent);
        modal = TestBed.get(NgbActiveModal);
        playerService = TestBed.get(PlayerService) as MockPlayerService;
        fixture.detectChanges();
    });

    it('displays error', () => {
        expect(fixture.nativeElement.querySelector('.error-message').textContent.trim()).toEqual('Something has gone wrong, going to try to re-login and reset.');
    });

    it('close modal', () => {
        expect(modal.dismiss).not.toHaveBeenCalled();
        fixture.nativeElement.querySelector('.close-button').click();
        expect(modal.dismiss).toHaveBeenCalled();
        expect(playerService.forceLogout).toHaveBeenCalled();
    });
});
