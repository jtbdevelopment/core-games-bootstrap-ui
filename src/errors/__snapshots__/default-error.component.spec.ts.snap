// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  default error component displays error 1`] = `
<ngbd-modal-content
  activeModal={[Function MockModal]}
  playerService={[Function MockPlayerService]}
>
  <div
    class="general-error-dialog"
    role="dialog"
  >
    
      
    <div
      class="modal-header"
    >
      
          
      <h4
        class="modal-title"
      >
        Sorry!
      </h4>
      
      
    </div>
    
      
    <div
      class="modal-body"
    >
      
          
      <span
        class="error-message"
      >
        Something has gone wrong, going to try to re-login and reset.
      </span>
      
      
    </div>
    
      
    <div
      class="modal-footer"
    >
      
          
      <button
        class="btn btn-info focus close-button"
      >
        OK
          
      </button>
      
      
    </div>
    
  
  </div>
  
</ngbd-modal-content>
`;
